File: named_args_test.rs
fn greet(name: &str, greeting: &str) -> String {
    return format!("{}, {}!", greeting, name);
}
fn add(a: i64, b: i64, c: i64) -> i64 {
    let result: i64 = (a + b) + c;
    return result;
}
fn main() -> () {
    let msg1: String = greet(&"World", &"Hello");
    let msg2: String = greet(&"Tane", &"Hi");
    let msg3: String = greet(&"User", &"Welcome");
    let msg4: String = greet(&"Friend", &"Hey");
    let sum1: i64 = add(1i64, 10i64, 20i64);
    let sum2: i64 = add(1i64, 2i64, 20i64);
    let sum3: i64 = add(1i64, 2i64, 3i64);
    let sum4: i64 = add(5i64, 15i64, 25i64);
    println!("{:?}", &msg1);
    println!("{:?}", &msg2);
    println!("{:?}", &msg3);
    println!("{:?}", &msg4);
    println!("{:?}", &sum1);
    println!("{:?}", &sum2);
    println!("{:?}", &sum3);
    println!("{:?}", &sum4);
}

File: examples/simple/named_args_test.py
# Test file for named arguments and default arguments

def greet(name: str, greeting: str = "Hello") -> str:
    return f"{greeting}, {name}!"

def add(a: int, b: int = 10, c: int = 20) -> int:
    result: int = a + b + c
    return result

def main() -> None:
    # Test default arguments
    msg1: str = greet("World")
    msg2: str = greet("Tane", "Hi")
    
    # Test named arguments
    msg3: str = greet(name="User", greeting="Welcome")
    msg4: str = greet(greeting="Hey", name="Friend")
    
    # Test with numeric defaults
    sum1: int = add(1)
    sum2: int = add(1, 2)
    sum3: int = add(1, 2, 3)
    sum4: int = add(a=5, b=15, c=25)
    
    print(msg1)
    print(msg2)
    print(msg3)
    print(msg4)
    print(sum1)
    print(sum2)
    print(sum3)
    print(sum4)

if __name__ == "__main__":
    main()
